server/jobs/checkpoint.py

```diff
@@ -0,0 +1,55 @@
+from sqlalchemy import or_
+from collections import defaultdict
+
+from server import jobs
+from server.models import Assignment, Score, Backup, db
+from server.utils import server_time_obj
+
+@jobs.background_job
+def assign_scores(assign_id, score, kind, message, deadline,
+                     include_backups=True):
+    logger = jobs.get_job_logger()
+    current_user = jobs.get_current_job().user
+
+    assignment = Assignment.query.get(assign_id)
+    students = [e.user_id for e in assignment.course.get_students()]
+    submission_time = server_time_obj(deadline, assignment.course)
+
+    # Find all submissions (or backups) before the deadline
+    backups = Backup.query.filter(
+        Backup.assignment_id == assign_id,
+        or_(Backup.created <= deadline, Backup.custom_submission_time <= deadline)
```

Comment:
I think the backup class that has some magic property that computes the real submission time. Would that work here?